pkg/kv/kvserver/rangefeed/processor.go

```diff
@@ -365,12 +399,18 @@ func (p *Processor) run(
 		// Close registrations and exit when signaled.
 		case pErr := <-p.stopC:
 			p.reg.DisconnectWithErr(all, pErr)
+			if p.memBudget != nil {
+				p.memBudget.Close(ctx)
```

Comment:
nit: to make these more concise, consider making `Close()` handle a `nil` receiver.

Reply:
Is a good practice in general to make some receivers nil safe and some unsafe?

Reply:
Eh, I think you'll get different answers depending on who you ask. I tend to do it when it's useful and avoids boilerplate, but don't have a strong opinion either way -- whatever your prefer is fine with me.

Reply:
Btw, maybe we should use a `defer` for this, to make sure it always gets cleaned up when we return from `run()`?